docs/migration-v3.md

```diff
@@ -15,6 +15,66 @@ Consult the [announcement post](/blog/announcement-v3) to learn more about the v
 npm add --save remirror@latest @remirror/react@latest @remirror/pm@latest
 ```
 
+## ES2023 decorators
+
+Remirror v3 has been updated to use ES2023 decorators, now they have [reached Stage 3](https://github.com/tc39/proposal-decorators), and [esbuild supports them](https://github.com/evanw/esbuild/releases/tag/v0.21.0). Previously Remirror used TypeScript's experimental decorators, which use an older version of the spec.
+
+If you use Remirror's decorators (`@extension`, `@command`, `@helper` and `@keyBinding`) in your custom extensions, you may find they no longer work. Here are a few things to try
+
+### TypeScript
+
+Set `experimentalDecorators` to **false** in your `tsconfig`. This ensures you don't use the older version of the decorator spec.
+
+### Vite
+
+If you are using Vite with the SWC plugin, see Next.js below, otherwise:
+
+Ensure the `esbuild` version you use is _at least_ `0.21.0`, and update the `target` your Vite config
+
+```json
+{
+  "esbuild": {
+    "target": "ES2020"
+  }
+}
+```
+
+### Babel
+
+Update your Babel config to use the latest version of the decorator spec.
+
+```json
+{
+  "plugins": [["@babel/plugin-proposal-decorators", { "version": "2023-11" }]]
+}
+```
+
+### Next.js
+
+At time of writing [SWC does not support the latest decorator syntax](https://github.com/vitejs/vite-plugin-react-swc/issues/86).
```

Comment:
Let's use this issue link instead of the one from `vite-plugin-react-swc`, as it's not directly related to Next.js.

https://github.com/vercel/next.js/issues/48360